dds/DCPS/transport/shmem/ShmemReceiveStrategy.cpp

```diff
@@ -88,8 +92,10 @@ ShmemReceiveStrategy::receive_bytes(iovec iov[],
         "(%P|%t) ShmemReceiveStrategy::receive_bytes link %@\n", link_));
 
   // check that the writer's shared memory is still available
-  ShmemAllocator* alloc = link_->peer_allocator();
   void* mem;
+
+  ShmemDataLink::PeerAllocatorProxy proxy(*link_);
+  ShmemAllocator* alloc = proxy.peer_allocator();
```

Comment:
Do these and the lines below that access `alloc` need to be in a scope similar to its use in the `read` function above?

Reply:
No.  It's used later in the function.